File: backend/src/main/java/com/ll/TeamProject/domain/schedule/dto/ScheduleResponseDto.java
package com.ll.TeamProject.domain.schedule.dto;

import com.ll.TeamProject.global.jpa.entity.Location;

import java.time.LocalDateTime;

public record ScheduleResponseDto(
        Long id,
        Long calendarId,
        String title,
        String description,
        LocalDateTime startTime,
        LocalDateTime endTime,
        Location location,
        LocalDateTime createDate,
        LocalDateTime modifyDate
) {
    public ScheduleResponseDto(
            Long id, Long calendarId, String title, String description,
            LocalDateTime startTime, LocalDateTime endTime, Location location,
            LocalDateTime createDate, LocalDateTime modifyDate
    ) {
        this.id = id;
        this.calendarId = calendarId;
        this.title = title;
        this.description = (description != null) ? description : ""; // null이면 빈 문자열 처리
        this.startTime = startTime;
        this.endTime = endTime;
        this.location = location;
        this.createDate = createDate;
        this.modifyDate = (modifyDate != null) ? modifyDate : createDate; // null이면 createDate와 동일하게 설정
    }
}

File: backend/src/main/java/com/ll/TeamProject/domain/schedule/dto/ScheduleResponseDto.kt
package com.ll.TeamProject.domain.schedule.dto;

import com.ll.TeamProject.global.jpa.entity.Location;
import org.springframework.data.jpa.domain.AbstractPersistable_.id

import java.time.LocalDateTime;

data class ScheduleResponseDto(
        val id:Long,
        val calendarId:Long,
        val title: String,
        val description: String="",
        val startTime: LocalDateTime,
        val endTime: LocalDateTime,
        val location: Location,
        val createDate: LocalDateTime,
        val modifyDate: LocalDateTime=createDate,
)
